src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java

```diff
@@ -1,31 +1,34 @@
 package uk.gov.hmcts.ccd.domain.model.search;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.databind.JsonNode;
 
+import java.util.HashMap;
 import java.util.Map;
 
 public class SearchResultViewItem {
     @JsonProperty("case_id")
     private String caseId;
     @JsonProperty("case_fields")
-    private Map<String, JsonNode> caseFields;
+    private final Map<String, Object> caseFields = new HashMap<>();
 
     public SearchResultViewItem() {
         // Default constructor for JSON mapper
     }
 
-    public SearchResultViewItem(final String caseId,
-                                final Map<String, JsonNode> caseFields) {
+    public SearchResultViewItem(final String caseId) {
         this.caseId = caseId;
-        this.caseFields = caseFields;
     }
 
     public String getCaseId() {
         return caseId;
     }
 
-    public Map<String, JsonNode> getCaseFields() {
+    public Map<String, Object> getCaseFields() {
         return caseFields;
     }
+
+    public <T> SearchResultViewItem addCaseFields(Map<String, T> caseFields) {
```

Comment:
another alternative which if we don't want to introduce generics would be:
```
    public SearchResultViewItem addCaseFields(Map<String, ? extends Object> caseFields) {

```
not sure which one is better though :)

Reply:
Its synonymous in this context, but wild card looks more readable. Changed now.